apps/web/src/pages/templates/editor/Preview.tsx

```diff
@@ -28,29 +32,41 @@ export const Preview = ({ activeStep, view }: { activeStep: number; view: string
     control,
   });
 
+  const variables = useWatch({
+    name: `steps.${activeStep}.template.variables`,
+    control,
+  });
+
   const { integrations = [] } = useIntegrations();
   const [integration, setIntegration]: any = useState(null);
   const [content, setContent] = useState<string>('<html><head></head><body><div></div></body></html>');
   const { isLoading, mutateAsync } = useMutation(previewEmail);
+  const processedVariables = useProcessVariables(variables);
+  const [payloadValue, setPayloadValue] = useState('{}');
 
   useEffect(() => {
-    if (contentType !== 'editor') {
-      if (!htmlContent) {
-        return;
-      }
-      setContent(htmlContent);
+    setPayloadValue(processedVariables);
+  }, [processedVariables, setPayloadValue]);
 
-      return;
-    }
-    mutateAsync({
-      content: editorContent,
-      contentType,
-    }).then((result: { html: string }) => {
+  const parseContent = (args: {
+    contentType: MessageTemplateContentType;
+    content: string | IEmailBlock[];
+    payload: any;
+  }) => {
+    mutateAsync(args).then((result: { html: string }) => {
       setContent(result.html);
 
       return result;
     });
-  }, [contentType, htmlContent, editorContent, setContent, mutateAsync]);
+  };
+
+  useEffect(() => {
+    parseContent({
+      contentType,
+      content: contentType === 'editor' ? editorContent : htmlContent,
```

Comment:
I think we can skip this argument `content` as we are sending the full `contentType`, and build the part of the payload that `mutateAsync` needs in line 56, therefore not having the dependencies for `editorContent` and `htmlContent`. Not sure if this could create a side effect of the effect not triggering if `editorContent` or `htmlContent` change individually, but I am assuming that as they are part of `control` and we are using `useWatch` that would be covered.

Reply:
Sorry about that. What I tried to say is that we can do this:
```
 const parseContent = ({
    contentType: MessageTemplateContentType;
    payload: any;
  }) => {
    const content = contentType === 'editor' ? editorContent : htmlContent;
    const args = {
       contentType,
       content,
       payload
     };
     
    mutateAsync(args).then((result: { html: string }) => {
      setContent(result.html);

      return result;
    });
  };
  ```
  So the useEffect will look like this:
  ```
  useEffect(() => {
    parseContent({
      contentType,
      payload: processedVariables,
    });
  }, [contentType, parseContent, processedVariables]);
  ```